modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -148,6 +162,54 @@ public void testRequestStats() throws Exception {
         super.testRequestStats();
     }
 
+    public void testAbortRequestStats() throws Exception {
+        final String repository = createRepository(randomRepositoryName());
+
+        final String index = "index-no-merges";
+        createIndex(index, 1, 0);
+        final long nbDocs = randomLongBetween(10_000L, 20_000L);
+        try (BackgroundIndexer indexer = new BackgroundIndexer(index, client(), (int) nbDocs)) {
+            waitForDocs(nbDocs, indexer);
+        }
+        flushAndRefresh(index);
+        ForceMergeResponse forceMerge = client().admin().indices().prepareForceMerge(index).setFlush(true).setMaxNumSegments(1).get();
+        assertThat(forceMerge.getSuccessfulShards(), equalTo(1));
+        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
+
+        // Intentionally fail snapshot to trigger abortMultipartUpload requests
+        shouldFailCompleteMultipartUploadRequest.set(true);
+        final String snapshot = "snapshot";
+        clusterAdmin().prepareCreateSnapshot(repository, snapshot).setWaitForCompletion(true).setIndices(index).get();
+        clusterAdmin().prepareDeleteSnapshot(repository, snapshot).get();
+
+        final RepositoryStats repositoryStats = StreamSupport.stream(
+            internalCluster().getInstances(RepositoriesService.class).spliterator(),
+            false
+        ).map(repositoriesService -> {
+            try {
+                return repositoriesService.repository(repository);
+            } catch (RepositoryMissingException e) {
+                return null;
+            }
+        }).filter(Objects::nonNull).map(Repository::stats).reduce(RepositoryStats::merge).get();
+
+        // Ignore PutMultipartObject for comparison since failed multipart upload requests (null response) are not counted for sdk stats
```

Comment:
Are we accounting for failed `PutMultipartObject` requests? Maybe we should structure the failure injection differently so we don't have to skip those